### DISCUSSION TITLE
2024-9-25 編集会議（日本語組版の基本アーキテクチャ、約物）

### DISCUSSION THREAD

**Discussion by kidayasuo** [w3c/jlreq-d#64] [General]
# 2024-9-25 編集会議
出席：敏先生、木田

- 内容の箇条書きではなく、テキストを実際にドラフトしてゆく方針で進め、3.1 日本語組版の基本アーキテクチャ、4.1.1 約物の配置の基本、4.1.2 括弧類・句読点等の空白を削除する例、をドラフトした。またこれらの章や節のタイトルを調整した。
- 前回の会議で4章にあった、プロポーショナル組み、および分かち書き、は11. 読みやすさと視覚効果、に移動。
- 3.2 ラテンベースラインシステムでの組版、については書く内容を10/1のJLReq TFで相談することとした。木田が案内を出す。
- 編集会議を頻繁に、1時間程度の長さで開くこと、次回の編集会議では第1章の先頭から見てゆくこと、を合意した。

---- 
## 3. 日本語組版の基本アーキテクチャ(Foundational Architecture of Japanese Line Layout)
> 詳細ではなくアーキテクチャの話。英語ベースの組版を知っている人に、それとは異なるアーキテクチャであることを印象付ける。また、ではラテンベースラインの仕組みの中にどのように実装するか、 を説明する。

### 3.1 日本語組版の基本アーキテクチャ
> ここではまず和字の基本を説明して、和字の組版は欧文ベースラインとは異なるシステムであることを説明。和字の組版の基本をわかってもらうためにある。本格的に説明すると言うよりも、ボディによる組み方のアーキテクチャ（中心揃え＋ベタ）を簡単に説明する。ベースラインという概念はないよ。サイズの異なる文字の配列について、中央揃えが基本だと説明。

> ラテン組版と日本語組版の違いをここでカバー。重要なポイント。
> 1. ボディが基本
> 2. 基本的にベタ組み（X方向）
> 3. どこでぶった斬っても良い（X方向）
> 4. 行内は中央揃えが緩い基本。88%を追認してもいい？？（Y方向）
> 5. 行間が違う（行と行との関係Y方向）

日本語の文字の設計は、統一したサイズの、多くの場合、正方形の枠、つまりボディがあって、その中で字面をデザインしている。ラテン文字のように、ベースライン、ミーンライン、アセント、ディセント、などの構造がない。（活字のイラスト、デジタル文字のイラスト）

多くの日本語フォントでは、ボディは正方形であり、同じグリフを縦組、横組両方で使用する。よって字面の設計にあたっては、縦組、横組両方の配置において文字間や上下左右の位置が揃って見えるようにデザインされている。ボディが長方形である日本語フォントもあり、縦長のものはもっぱら横組、横長のものはもっぱら縦組に使われる。

X方向（文字を並べてゆく方向）については、全幅のボディを密着させて配置していく。行を分割する場合、ラテン文字では単語間で行うが、日本語においては原則としてどこの文字間でも分割が可能である。

この基本のままでレイアウトできるのが最も理想的だが、下のような処理に対処するため、字間の調整の必要が出てくる。この処理の巧拙が日本語組版処理の読みやすさ、美しさに大きく影響する。この点について4章で詳説する。
1. 連続した約物の処理
2. 例外的に分割できない文字の処理
3. ラテン文字などプロポーショナルな文字の挿入
4. ジャスティフィケーションに伴う行の調整
5. 空け組、詰め組の処理（トラッキングの処理）

一行の中ではY方向はボディを揃えてゆけば良い。文字サイズが異なる場合は、中心で揃えるのが基本であるが、場合によって横組では下端、縦組では右端を揃えることもある。

日本語の行間は一般的に英語よりも広く取る必要がある。これは、殆どの文字が枠いっぱいにデザインされているからである。これは英語において、多くの文字の字面がベースラインとミーンラインの間に収まっているため、ディセンダーやアセンダーのミーンラインより上の部分が実質的に空間となっているためである。

### 3.2 ラテンベースラインシステムでの組版を説明
> ここは、JLReq TFのミーティングで議論する。
> 予告を出して、準備しておいてもらう。
> 木田：山本さんの文章およびwebの規格を読んでおく。

> 早い目にベースラインの中にどう組み込むかの現実的な話に持って行く。なぜなら、それがエンジニアが実際に対処しなければならない状況だから。混植は別の話。

多くの現状：簡便な方法としては、日本語文字のボディの中に、ラテンベースラインの位置を決め、それに基づいて、ラテンベースラインのシステムの上で日本語を配置していく。この場合、異なる文字サイズの日本語を並べる場合に位置が揃わない問題がある（図示）。英語が主な場合に良いかも？

より良い方法：ベースラインを複数持つ方法？

英文を主に考えるか日本語を主に考えるか、を分ける必要のない、両方を高度にサポートできる統合された方法はないか？

和欧混植については4.2で説明する。

## 4. 行への文字の配置の実際

前章で述べたように、全幅等幅での配列が理想であるが、字間の調整の必要な場合がある。この処理の巧拙が日本語組版処理の読みやすさ、美しさに大きく影響する。

### 4.1.1 約物の配置の基本

和文用の括弧類や句読点は，一般に字幅の半分よりやや小さい字面を持っており，その字幅の半分程度の空白がある．この空白は、始め括弧類では前側，終わり括弧類や句読点では後ろ側にある．中点類は前後に字幅の1/4ずつの空白がある．よって、その全幅のボディを密着させて配置していけば空白は確保される．

注：日本語の中に使用する括弧類や句読点，中点類は，ラテン文字用の括弧などがある例もある．特に必要があるために使用する場合を除外し，和文用の括弧類，句読点，中点類を使用する．たとえば，“索引（index）を”とあった場合も，たまたま小括弧内の内容がラテン文字であるが，小括弧は和文と考える．

### 4.1.2 括弧類・句読点等の空白を削除する例

次のようなケースでは，括弧類や句読点などの空白を削除、または幅の調整をする処理が必要になる．

- 括弧類や句読点を連続して配置する場合
- 括弧類が行頭に配置する場合
- 行の調整処理をする場合

また、下の文脈の場合、この空白を削除する場合がある。（例）
- 漢数字の間に読点（U+3001）を用いる場合
- 中点（U+30FB）で漢数字の小数点を示す場合
- 文中に現れる小括弧などの空白を削除して配置する方法がある

> 以降、約物の部分はdraftsにある内容をベースに編集する
## 7. 縦組
- 組方向の変更
- 縦組と横組で字形等が異なる例
- 数字の表記と組版処理
- ラテン文字等の処理
- 縦組と横組の句読点
- 縦組と横組の括弧類
- 縦組と横組で注意が必要な行処理等
- 縦組と横組で異なる注等の配置処理

## 8. 段落の処理（複数の行の処理）
- 行揃えの選択
- 段落の示し方
- 行長
- 行間
- 箇条書き

## 9 見出し
> 書くことあまりないけど、なければ、日本語独自のものがない、と書く。

### 11. 2 文字サイズに関わる問題
- 大きいサイズの文字の処理（ここに詰め組も来る）
- 小さいサイズの文字の処理（書くことある？）
- オプティカルサイジング（サイズに適した書体）

**Comment by KobayashiToshi** [comment]
> 3.2 ラテンベースラインシステムでの組版を説明

の内容ですけれど，ここでは，Y方向の処理において，ラテンベースラインシステムでは，ベースラインという基準を元に処理するが，和文では，そうはいかない．ボディを元に横組では，3つの位置，それとは別に縦組でも3つの位置の基準位置が必要になる．

という違いがあることだけを記述するだけでいいのではないかな．

こうした違いがあるラテン文字と和文が，同一行に混ぜて配置する場合の問題点は，別項で解説する，という追記だけしておいて，この問題は別項で述べるということで，どうでしょうか？

**Reply by kidayasuo** [reply]
はい、ここは単純にして、ラテン文字との配置の問題は別項で良いと思います。この節で最も重要なことは、「箱を並べる」考え方から、「基準線に沿って並べてゆく」という考え方への変換です。その理由は、現在の知られているすべてのシステム（InDesign日本語版を除いて）が、基準線に沿って並べるという考えで作られているからです。

敏先生の言われる「3つの基準位置」は、三種類の基準線（ベースライン）によって表されます。欧文が使う、アルファベティック・ベースライン（CSSはその用語を使っていますので、それに合わせましょう）との違いは、Y方向の位置です。CSSも、ドラフトにおいて、多くの種類のベースラインを定義してます。https://www.w3.org/TR/css-inline-3/#css-metrics

もう一点、以降で詳しく説明する日本語のレイアウトにおいて、ボティの上端下端（横書きの場合）の位置が必要になった場合の、その見つけ方を書いておいた方が良いように思います。考え方を書いて、サイドノートに、OpenTypeでのやり方を書くのが良いでしょうか。

以降の日本語レイアウトの記述はすべて「基準線に沿って並べてゆく」考えで説明するのが良いと思いますが、和文だけを考慮する場合は依然としてボティを使って、欧文が絡むところは線に沿う考え方にスイッチする、という考えもできます。ここは編集会議で議論して、次回のJLReq TFで最終的に決めるのはどうでしょう。

**Comment by KobayashiToshi** [comment]
> ボティの上端下端（横書きの場合）の位置が必要になった場合の、その見つけ方を書いておいた方が良いように思います。

それはいいことだと思いますが，ラテン文字の場合，デセンダーの下端からアセンダーの上端までで，とりあえずボディの大きさと考えてよいのかな？　つまり，X方向はサイドベアリングを考えないといけないが，Y方向は，サイドベアリングを考えなくとよい．なぜならX方向はベタがデフォルトなのでサイドベアリングを考えないといけないが，Y方向は，原則としてベタ（つまり行間ゼロ）は，デフォルトではなく，多くは何らかのアキを確保するので，サイドベアリングを考えなくとよい．

> この節で最も重要なことは、「箱を並べる」考え方から、「基準線に沿って並べてゆく」という考え方への変換です。

そうかな？　この節では，“基準線に沿って並べてゆく”場合，Y方向では，基準となる位置の考え方，和文文字とラテン文字では異なるという点はないですか？　なお，以下では，箱を並べる→アキ方式　基準線に沿って並べてゆく→送り方式　と呼ぶことにします．

つまり，論点は，2つあり，
-  アキ方式か送り方式か　この問題は，ラテン文字でも和文文字でも共通
-  アキ方式でも送り方式であっても，同一行に異なる文字サイズの文字を並べる場合，その基準位置の取り方の考え方に違いがある．

この2つは分けて考えた方がよい．3.2は後者だけでよい．アキ方式か送り方式かの問題は，別項で解説することではないか？

そして，前者のアキ方式か送り方式かは，いろいろと考えないといけない問題のように思います．　

1　何を重視するか
実際に合わせるという点では送り方式です．
実際に見た目の組版結果を示すという点では，アキ方式の方が，特に日本語組版ではいいように思います．ことにＸ方向は，そうです．みんさん，ラテン文字とのアキも四分とか八分という言い方をしています．Ｙ方向も，私はアキ方式の方が考えやすい．

2　簡単に換算できるからという考え方
アキ方式も送り方式も，換算は計算できます．同一文字サイズの場合は，どこに基準位置があっても問題はありません．ただ，問題は，異なる文字サイズが混じる場合です．この場合，基準位置の場所により送りの値は変わってきます．これに対し，アキ方式では，ボディを基準に考えるという点では，異なる文字サイズが混じる場合でも誤解を与えることは少ないでしょう．

3　ケースにより説明を変える，あるいは，両方を併記する
現在は，X方向はアキ方式で説明しています．Y方向については，X方向ほど説明することは多くありません．
あるいは，アキ方式と送り方式の原則的な考え方をどこかで説明するよいうことも考えられるかもしれません．

4　Ｘ方向とY方向では，どちらが考えやすいか．X方向では，ラテン文字でも和文文字でも共通して，ボディ間のアキが問題となるので，アキ方式が考えやすい．では，Y方向では，どうか．

もっとも，Y方向の行間については，パーセント又は分数で，以下のように書けば，併記もできる．
　二分行間　→　150％，つまりアキは50％　2/3，つまりアキは1/2　
　四分行間　→　150％，つまりアキは50％　5/4，つまりアキは1/4

字間は分数がなじみやすいが，行間は，私は分数がなじみやすいが，一般の人にとってはパーセントの方がよいかもしれない．いっそのこと，字間も，行間もパーセントで書いてしまうということも考えられる．

いずれにしても，ということでいいと思います．
> ここは編集会議で議論して、次回のJLReq TFで最終的に決めるのはどうでしょう。

**Reply by kidayasuo** [reply]
すみません、アキ方式、送り方式、がよくわかっていません。例えば、文字幅の後ろにその1/4のアキを空けることを、1/4のアキと表現するか、125%送るというか、という違いですか？

X方向側は、実装を見ていると問題は比較的に少ないと思います。そもそも現実のフォントが、日本語であろうと英語であろうと同じ仕組みで作られているので、まずはそこで違いが吸収されているからかもしれません。字間の設定は、CSSでは、letter-spacingはemが単位として使えますから、25%とすると四分アキになります。ワープロなどのトラッキングの設定も同じような感じです。和欧間などの実装はこれからですが、1/6, 1/8などと表現しても、混乱はないんじゃないかな。そもそも欧米はむしろ日本よりも分数で数を表すことが多いですし。

**Comment by KobayashiToshi** [comment]
> アキ方式、送り方式

9ポでベタ組にする場合，以下のようになる．これはX方向もY方向も同じ．
　アキ方式→ベタ　送り方式→送りは9ポ
9ポで二分アキにする場合，以下のようになる
　アキ方式→二分アキ又は4ポアキ　送り方式→送りは13.5ポ
つまり，アキ方式はボディのサイズがデータとして必要であり，そのボディのサイズから考える．アキ方式は，ボディ間のアキで示す．送り方式では，文字サイズは前提となるが，基準点があきらかであれば，ボディの大きさは分からなくてもよい．なぜなら，送り量は前の文字の基準点から次の文字の基準点までの距離であるから．

詰め組でのアキ方式は，やや難点があるが，マイナスの値にすれば切り抜けられる．詰め組の場合，送り方式では，送り量を文字サイズより数値を小さくすればよい．トラッキングの場合，同様に考えてよいが，一般にはボディ間のアキをマイナスの相対量で示すからアキ方式といえる．

この問題とは別に長さの示し方の問題がある．従来は，一般に数値（5ポ，8ポなど）で指定する場合と相対量（二分，四分）があったが，この相対量は分数でもパーセントで示しても同じである．ただ，使われていなかっただけ．相対量は分数でもパーセントの方が，組版の専門家以外の人には，なじみやすいかもしれない，ということ．

ところで，3.2は，以下のように議論を展開できる．こうすれば，以下のように分けなくて説明できる．この説明の方が木田さんのいいたいことに近いのかもしれない．ただ，説明文は，冗長で整理されていません．
 
>    アキ方式か送り方式か　この問題は，ラテン文字でも和文文字でも共通
>   アキ方式でも送り方式であっても，同一行に異なる文字サイズの文字を並べる場合，その基準位置の取り方の考え方に違いがある．

1　X方向の文字間の配置位置は，実際の処理は送り方式であるが，説明や組版の指示では，アキ方式でも送り方式でもよいが，一般にアキ方式で示すことが多い．これは，和文文字もラテン文字も基準点と字幅情報から，ボディのサイズを簡単に認識でき，しかも，X方向はボディのアキ（マイナスの値を含む）が主に問題になるからである．

2　Ｙ方向は，実際の処理は送り方式であるが，説明や組版の指示では，行間を考える場合，アキ方式でも送り方式でもよい．しかし，同一行に異なるサイズの文字が混ざる場合では，和文文字とラテン文字では異なる．
- 和文文字：通常はボディの中心でそろえるが，ケースによっては，ボディのＹ方向の前の端又は後ろの端でそろえる場合もある．したがって，和文ではボディのサイズが配置の基準になる．このようにボディのサイズが基準となるのであれば，それとの整合性から，行間もアキ方式の方がよいことになる．ただし，アキ方式でも送り方式でも簡単に換算できるから，どちらでもよいといえる．ただ，領域の先頭または末尾に配置する場合，領域との空白を考えると，ボディのサイズが配置の基準の方が説明は簡単であり，その意味では，アキ方式がいいかもしれない．もっとも，ボディを基準にしないでも，Y方向の基準点として，ボディの中心と両端のデータがあれば，送り方式，つまる基準点方式でもよいことになる．
-  ラテン文字では，異なるサイズの文字が混ざる場合でも，ベースラインを基準に考えればよいので，送り方式，つまり基準点を元に配置していく方式がよいことになる．しかも，ラテン文字では，Ｙ方向のボディの大きさのデータは不可欠のものではなく，しかもボディの大きさそのものがあいまいであるという問題がある．

このように，Y方向については，ベースラインの基準点を元に処理するラテン文字とボディのサイズ又ラテン文字とは異なる基準点を必要とする和文文字が混ざる場合の問題点は，別項で解説する．
